packages/core/src/fetchers/fetchReferencedEntities.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { ContentfulClientApi, Entry } from 'contentful';
+import { fetchReferencedEntities } from './fetchReferencedEntities';
+import { compositionEntry } from '../test/__fixtures__/composition';
+import { assets, entries } from '../test/__fixtures__/entities';
+import { describe, it, expect, vi, Mock } from 'vitest';
+
+const mockClient = {
+  getEntries: vi.fn(),
+  getAssets: vi.fn(),
+} as unknown as ContentfulClientApi<undefined>;
+
+describe('fetchReferencedEntities', () => {
+  it('should throw an error if client has not been provided', async () => {
+    try {
+      await fetchReferencedEntities({
+        // @ts-expect-error intentionally setting it to undefined
+        client: undefined,
+        experienceEntry: compositionEntry,
+        locale: 'en-US',
+      });
+    } catch (e) {
+      expect((e as Error).message).toBe(
+        'Failed to fetch experience entities. Required "client" parameter was not provided'
+      );
+    }
```

Comment:
[request change]
Same feedback as for the tets in `fetchExperienceEntry.spec.ts` that try/catch clauses are not working for jest tests.